plugins/inputs/ctrlx_datalayer_sse/README.md

```diff
@@ -0,0 +1,139 @@
+# ctrlx_datalayer_sse Input Plugin
+
+The `ctrlx_datalayer_sse` plugin gathers data from the ctrlX Data Layer which is the communication middleware runnning on the ctrlX CORE runtime of ctrlX CORE devices from [Bosch Rexroth](https://boschrexroth.com). The ctrlX CORE is usually used for professional automation applications like industrial automation, building automation, robotics, IoT Gateways or as classical PLC. For more information, see [ctrlX AUTOMATOIN](https://ctrlx-automation.com).
```

Comment:
Ok, but will there be other plugins upcoming for the other interfaces? And if so, wouldn't it make sense to handle all of them in the same _Telegraf_ plugin?
One reason why I want you to think about the name is because first I'm lazy when it comes to typing ;-) and second it will be printed for every log, so you see a loooong name and the actual message is way to the right...

Reply:
We talked a lot about a different naming but could not come up with a satisfying alternative. The potential future protocols will probably be based on a total different technology, that we do not want to mix with this plugin. 

**Regarding the `_listener` suffix:**
Can you please give us a link to the mentioned naming conventions about the `_listener` suffix? We looked at the `execd` plugin which seems to use also the service interface but is not having a `_listener` suffix. Also we think, that the suffix has no benefit for the users, because it is only an implementation detail. 

So at the moment `ctrlx_datalayer_sse` is still our favourite and is not too long. What do you think?

Reply:
It seems like we indeed do not have this documented... :-/ Anyway, most of the plugins adhere to this convention of having a `_listener` (e.g. [KNX](https://github.com/influxdata/telegraf/tree/master/plugins/inputs/knx_listener)) or `_consumer` (e.g. [MQTT](https://github.com/influxdata/telegraf/tree/master/plugins/inputs/mqtt_consumer)) suffix.
This makes it easier for users and maintainers to directly see that the plugin is not _actively_ gathering the data but is rather waiting for data to be pushed.

Can you elaborate a bit on what additional plugin names would look like?

Reply:
In that definition, our plugin is similar to MQTT (consumer). Our plugin is not listening for arbitrary messages to be pushed by a client. There is no listener port that is opened by the plugin.
The plugin itself is a client, that is connecting to a server (device). It is just for the sake of communication performance, that the client (plugin) is not polling the data, but the server is pushing further events on value change. But that's in our opinion only an implementation detail and not changing how the values are configured by the user (still a client configuration). 

This would mean: `ctrlx_datalayer_sse_consumer` ? What do you think?

Potential future protocols (low priority): 
- `ctrlx_datalayer_tcp_consumer`
- `ctrlx_drive_sip`